### PR TITLE
Fix: Add mechansim to check if a keyword is a valid PYACTION keyword for being inserted

### PR DESCRIPTION
**Pull request by lisajulia**
Before, there was no such check, I've added all keywords I've tested so far to the valid list, and I'll keep adding more keywords after having tested them.

Also, I've added the keyword "WELTARG" from https://github.com/OPM/opm-simulators/blob/master/python/test/test_schedule.py
@hakonhagland: Did you do more tests with the "insert_keywords" function for other keywords that are not in this [list](https://github.com/OPM/opm-common/pull/4008/files#diff-c473c725418b8f6963c6e2b19371ebaab58f49f41334deef2b666082e7637740R40) yet?
If so, I will add the respective keywords!

### CODE REVIEW

File: opm/input/eclipse/Schedule/Action/PyAction.cpp [hunk 1]
```diff
@@ -37,6 +37,19 @@ namespace py = pybind11;
...
+    static std::unordered_set<std::string> pyaction_allowed_list = {
+        "BOX",
+        "FIELD",
+        "ENDBOX",
+        "GCONPROD",
+        "NEXTSTEP",
+        "METRIC", "MULTX", "MULTX-", "MULTY", "MULTY-", "MULTZ", "MULTZ-",
+        "WCONPROD", "WEFAC", "WELOPEN"
+    };
+    return (pyaction_allowed_list.find(keyword) != pyaction_allowed_list.end());
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Action/PyAction.cpp:50]: unnecessary parens

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 2]
```diff
@@ -1420,18 +1420,23 @@ File {} line {}.)", pattern, location.keyword, location.filename, location.linen
...
-                                *keyword,
-                                parseContext,
-                                errors,
-                                grid,
-                                matching_wells,
-                                /*actionx_mode=*/false,
-                                &sim_update,
-                                &target_wellpi,
-                                wpimult_global_factor);
+            if (Action::PyAction::valid_keyword(keyword->name())){
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Schedule.cpp:1423]: missing space before brace

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 3]
```diff
@@ -1420,18 +1420,23 @@ File {} line {}.)", pattern, location.keyword, location.filename, location.linen
...
+                                    parseContext,
+                                    errors,
+                                    grid,
+                                    matching_wells,
+                                    /*actionx_mode=*/false,
+                                    &sim_update,
+                                    &target_wellpi,
+                                    wpimult_global_factor);    
+            } else {
+                std::string msg_fmt = fmt::format("The keyword {} is not supported for inserting it in a PYACTION", keyword->name());
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Schedule.cpp:1437]: const

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 4]
```diff
@@ -1420,18 +1420,23 @@ File {} line {}.)", pattern, location.keyword, location.filename, location.linen
...
+                parseContext.handleError( ParseContext::PYACTION_ILLEGAL_KEYWORD, msg_fmt, keyword->location(), errors);
```
Comment (akva2) [inline opm/input/eclipse/Schedule/Schedule.cpp:1438]: spurious space after parens

**Review by akva2 — APPROVED** [review]
Just some cosmetics

**Comment by lisajulia** [comment]
jenkins build this please

**Comment by blattms** [comment]
Seems like Jenkins died...
Could it be that the memory consumption increased again?

**Comment by blattms** [comment]
jenkins build this please

**Comment by lisajulia** [comment]
jenkins build this please

**Comment by lisajulia** [comment]
jenkins build this please

**Comment by lisajulia** [comment]
jenkins build this please

File: opm/input/eclipse/Python/PyRunModule.cpp [hunk 5]
```diff
@@ -144,7 +144,16 @@ help(opm_embedded.current_summary_state)
...
             return this->executeInnerRunFunction(actionx_callback);            
         } else {
-            this->module.reload();
+            try {
+                this->module.reload();
+            } catch (const std::exception& e) {
+                OpmLog::error(fmt::format("Exception thrown in python module {}: {}", this->module_name, e.what()));
+                throw e;
+            } catch(...) {
+                auto errorMessage = fmt::format("General exception thrown in python module {}", this->module_name);
```
Comment (akva2) [inline opm/input/eclipse/Python/PyRunModule.cpp:153]: you can use OPM_THROW here.

File: tests/msim/test_msim_ACTIONX.cpp [hunk 6]
```diff
@@ -621,6 +621,17 @@ BOOST_AUTO_TEST_CASE(PYTHON_CHANGING_SCHEUDULE) {
         }
     }
 }
+BOOST_AUTO_TEST_CASE(MSIM_PYACTION_INSERT_INVALID_KEYWORD) {
```
Comment (akva2) [inline tests/msim/test_msim_ACTIONX.cpp:624]: newline

**Review by akva2 — APPROVED** [review]
Two small niggles

File: opm/input/eclipse/Python/PyRunModule.cpp [hunk 7]
```diff
@@ -81,7 +82,7 @@ bool PyRunModule::executeInnerRunFunction(const std::function<void(const std::st
     } catch(...) {
         auto errorMessage = fmt::format("General exception thrown when calling run(ecl_state, schedule, report_step, summary_state, actionx_callback) function of {}", this->module_name);
         OpmLog::error(errorMessage);
-        throw std::runtime_error(errorMessage);
+        OPM_THROW(std::runtime_error, errorMessage);
```
Comment (akva2) [inline opm/input/eclipse/Python/PyRunModule.cpp:85]: ah I should have been more explicit. The whole point of using OPM_THROW is that it also puts the error in the log files.

Comment (lisajulia) [reply opm/input/eclipse/Python/PyRunModule.cpp:85]: yes, I just saw that :D and was about to change it

**Comment by lisajulia** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Comment by blattms** [comment]
Approved already. Merging.

**Pull request merged**